jmclient/jmclient/maker.py

```diff
@@ -147,7 +146,8 @@ def verify_unsigned_tx(self, txd, offerinfo):
         times_seen_cj_addr = 0
         times_seen_change_addr = 0
         for outs in txd['outs']:
-            addr = btc.script_to_address(outs['script'], get_p2sh_vbyte())
+            #FIXME: the type of address should be detected from the script (p2pkh/p2sh)
+            addr = self.wallet.script_to_address(outs['script'])
```

Comment:
It would be easy to write a routine to detect P2PKH scripts instead of blindly passing to the wallet and hoping it will be correct (in all standard use cases it will be). I could add that if requested.

Reply:
this code crashes when doing a non-sw cj with at least one p2sh output. will push a fix as soon as I manage to test that fix.